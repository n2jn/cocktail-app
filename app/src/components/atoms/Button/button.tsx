import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import gStyles from '~gStyles';

type AcceptedButtonType =
  | 'error'
  | 'success'
  | 'warning'
  | 'disabled'
  | 'default';

type ButtonProps = TouchableOpacityProps & {
  text: string;
  type?: AcceptedButtonType;
  isDisabled?: boolean;
};

export const Button = ({
  text,
  type = 'default',
  isDisabled,
  ...buttonProps
}: ButtonProps) => {
  const buttonStyle = buttonTypeStyles[type];
  const textStyle = textTypeStyles[type];

  return (
    <TouchableOpacity
      disabled={isDisabled}
      style={[gStyles.pv8, gStyles.mh16, styles.container, buttonStyle]}
      {...buttonProps}>
      <Text style={[styles.text, gStyles.text.bold16, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};

const buttonTypeStyles = StyleSheet.create({
  disabled: {
    backgroundColor: 'grey',
  },
  error: {
    backgroundColor: 'red',
  },
  warning: {
    backgroundColor: 'orange',
  },
  success: {
    backgroundColor: 'green',
  },
  default: {
    backgroundColor: 'black',
  },
});

const textTypeStyles = StyleSheet.create({
  disabled: {},
  error: {},
  warning: {},
  success: {},
  default: {},
});

const styles = StyleSheet.create({
  container: {
    borderRadius: 20,
    borderWidth: 2,
    backgroundColor: 'transparent',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
    textAlign: 'center',
  },
});
