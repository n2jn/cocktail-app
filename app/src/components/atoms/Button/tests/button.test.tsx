import {Button} from '../button';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react-native';

describe('basic test', () => {
  test('Button renders correctly', () => {
    render(<Button text={''} />);
    expect(screen).toMatchSnapshot();
  });
});

describe('type tests', () => {
  test('Should render warning type correctly', () => {
    const tree = renderer
      .create(<Button type={'warning'} text={''} />)
      .toJSON();
    expect(tree).toHaveStyle({backgroundColor: 'orange'});
  });

  test('Should render error type correctly', () => {
    const tree = renderer.create(<Button type={'error'} text={''} />).toJSON();
    expect(tree).toHaveStyle({backgroundColor: 'red'});
  });

  test('Should render success type correctly', () => {
    const tree = renderer
      .create(<Button text={''} type={'success'} />)
      .toJSON();
    expect(tree).toHaveStyle({backgroundColor: 'green'});
  });
});
