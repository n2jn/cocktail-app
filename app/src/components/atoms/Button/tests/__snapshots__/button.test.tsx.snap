// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test Button renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "black",
      "borderRadius": 20,
      "borderWidth": 2,
      "justifyContent": "center",
      "marginHorizontal": 38,
      "opacity": 1,
      "paddingVertical": 19,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "textAlign": "center",
        },
        Object {
          "fontFamily": "Cochin",
          "fontSize": 38,
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  />
</View>
`;
